ostd/src/task/scheduler/mod.rs

```diff
@@ -0,0 +1,273 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Scheduling subsystem (in-OSTD part).
+//!
+//! This module defines what OSTD expects from a scheduling implementation
+//! and provides useful functions for controlling the execution flow.
+
+mod fifo_scheduler;
+
+use core::sync::atomic::{AtomicBool, Ordering};
+
+pub(crate) use fifo_scheduler::FifoScheduler;
+use spin::Once;
+
+use super::{processor, task::Task};
+use crate::{
+    arch::timer,
+    cpu::{local::PREEMPT_INFO, this_cpu},
+    prelude::*,
+};
+
+static SCHEDULER: Once<&'static dyn Scheduler<Task>> = Once::new();
+
+static SHOULD_PREEMPT: ShouldPreemptFlag = ShouldPreemptFlag::new();
```

Comment:
Move this global variable to the end of this file, close to where `ShouldPreemptFlag` is defined.